### PR TITLE
Added install skills UI

### PR DESCRIPTION
**Pull request by nikiatanasow**

### CODE REVIEW

File: bot/default-skills.json [hunk 1]
```diff
@@ -0,0 +1,6 @@
+[
+    {
+        "name": "teambot-hello",
+        "source": "teambot-hello@https://github.com/teambot-club/teambot-hello"
```
Comment (angeltsvetkov) [inline bot/default-skills.json]: How about use NPM version instead of GitHUb one

File: bot/skills-loader.js [hunk 2]
```diff
@@ -1,16 +1,17 @@
-var SkillsLoader = function() {
+var SkillsLoader = function () {
     var npm = require('npm'),
+        skillsProvider = require('../server/providers/skills-provider'),
```
Comment (angeltsvetkov) [inline bot/skills-loader.js]: Use absolute path instead

File: server/providers/skills-provider.js [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+            dataProvider.remove({ 'name': skillName }, skillsCollectionName, callback);
+        } catch (err) {
+            callback(err);
+        }
+    }
+
+    function upsertSkills(skills, callback) {
+        try {
+            skills.forEach(function (skill) {
+                dataProvider.update({name: skill.name}, skill, skillsCollectionName, function () {
```
Comment (yyosifov) [inline server/providers/skills-provider.js]: I think an incorrect behavior might be present here. Foreach would be a Synchronous function, which will trigger N async operations. Before the Async Operations are finished, you will be out of the forEach loop and you'll call the callback(), which might not be the expected behavior. The callback function here is unused, too.

File: bot/skills-loader.js [hunk 4]
```diff
@@ -81,30 +82,45 @@ var SkillsLoader = function() {
     }
 
     function installPredefinedSkills(controller, middleware, devSkill, callback) {
```
Comment (angeltsvetkov) [inline bot/skills-loader.js]: The third argument should be 'skill' only. I know this is not part of your change

File: server/index.js [hunk 5]
```diff
@@ -49,6 +49,8 @@ exports.start = function(port) {
 
     server.delete('/team', teamsService.removeTeamInfo);
 
+    
+    server.post('/install-skill', botService.installSkill);
```
Comment (angeltsvetkov) [inline server/index.js]: From my perspective this should be /skill/install

File: portal/controllers/home.js [hunk 6]
```diff
@@ -1,29 +1,54 @@
...
             });
         };
 
+        
+        $scope.resetSettings = function() {
+            var confirm = $mdDialog.confirm()
+                .clickOutsideToClose(true)
+                .title('Reset Teambot settings?')
+                .textContent('All data will be lost.')
+                .ariaLabel('Lucky day')
```
Comment (DimitarTachev) [inline portal/controllers/home.js]: ?

**Review by theqaangel — APPROVED** [review]

**Pull request merged**